25_exceptions_lab/03.py

```diff
@@ -1,4 +1,35 @@
-"""
-Write code to make the following unit test pass
-"""
-
+"""
+Write code to make the following unit test pass
+"""
+
+import unittest
+import re
+import sys
+
+class InvalidImageExt(Exception):
+
+    def __init__(self, file):
+        super(InvalidImageExt, self).__init__("File '{}' has an invalid image extension".format(file))
+
+
+class ImageFile(object):
+
+    def __init__(self, file):
+        if not re.search(r'[.]gif|png|jpg|jpeg$', file):

```

Comment:
Python has a `os.path.basename` function which does a better job than writing your own regexp.

Regarding the regexp notice you need parenthesis around extensions because $ only refers to the last alternative (for example a.giff matches your regex -  try `re.search(r'[.]gif|png|jpg|jpeg$', '.giff')`